src/executor.rs

```diff
@@ -198,13 +198,69 @@ pub struct CoreVersionRef<'a> {
     pub state_version: Option<u8>,
 }
 
+impl<'a> CoreVersionRef<'a> {
+    /// Returns the SCALE encoding of this data structure.
+    pub fn scale_encoding_vec(&self) -> Vec<u8> {
+        let mut out = Vec::<u8>::with_capacity(
+            1 + self.spec_name.len() + 1 + self.impl_name.len() + 4 + 4 + 4 + 32 + 4 + 1,
```

Comment:
where 4, 32, 1 are coming from? should we maybe add a spec link or document?

Reply:
It's the lengths of the various fields right below.